### DISCUSSION TITLE
Wanted to contribute in Robolectric for GSoC 23', please help.

### DISCUSSION THREAD

**Discussion by arpitmx** [robolectric/robolectric#7904] [General]
Hey there👋,
My name is Alok Ranjan and I am a computer science student at JSS Noida from India. I am proficient in Java and have been doing Android development for more than three years using both Java and Kotlin. 

In addition to my experience with Android development, I am also interested in learning about compilers and have developed several apps and a library to solve various problems. I have published one app on the Google Play Store, and another is in the works.

I want to contribute to Robolectric for GSOC '23 , and have used Robolectric for unit testing before and I love it, so I have a personal zing for improving this project and really want learn about developing developer tools through the process! I will come up with creative ideas once I get involved. I will be grateful to get a chance to become a part of Robolectric community and will love to contribute to it even after GSoC period.

I am excited to start my journey with Robolectric, please help me out in getting started.

**Comment by utzcoz** [comment]
@arpitmx You can follow the official timeline of GSoC 2023 from https://summerofcode.withgoogle.com/programs/2023, and the Robolectric team will release potential topics if we apply and are accepted by GSoC team this year. 

>  and have used Robolectric for unit testing before and I love it

If you have used Robolectric before, you can start to contribute to Robolectric now if you have free time. There are some 'help wanted` issues in Robolectric commity: https://github.com/robolectric/robolectric/issues?q=is%3Aissue+is%3Aopen+label%3A%22help+wanted%22 that you can try.

> I will come up with creative ideas once I get involved.

New ideas are welcome for GSoC, and you also can post it to discuss out of GSoC.

> I am excited to start my journey with Robolectric, please help me out in getting started.

I think the best approach to contribute Robolectric with code is to use it, find bugs/features, try to fix or implement it and learn core modules of Robolectric, send PRs to review and learn from discuss, and PRs are merged. If you want to contribute to shadow implementations, the largest part of Robolectric, it's better you can read AOSP source code and implement some similar logic with pure Java for Robolectric.

**Comment by utzcoz** [comment]
The Robolectric will not participate in GSoC 2023.
